src/java.base/share/classes/sun/security/tools/keytool/Main.java

```diff
@@ -1922,33 +1965,80 @@ private void doGenKeyPair(String alias, String dname, String keyAlgName,
         }
 
         PrivateKey privKey = keypair.getPrivateKey();
+        CertificateExtensions ext;
 
-        CertificateExtensions ext = createV3Extensions(
-                null,
-                null,
-                v3ext,
-                keypair.getPublicKeyAnyway(),
-                null);
+        if (signerFlag) {
+            Certificate signerCert = keyStore.getCertificate(signerAlias);
+
+            X509CertImpl certImpl;
+            if (signerCert instanceof X509CertImpl) {
+                    certImpl = (X509CertImpl) signerCert;
+            } else {
+                certImpl = new X509CertImpl(signerCert.getEncoded());
+            }
```

Comment:
The exact same 7 lines above also appears in the code change above. Is it possible to combine them?

Reply:
Here is mainly to locate the signer’s SKID, and moved it up to the first occurrence of these lines of code to combine them.